designs/ami-minimum-age.md

```diff
@@ -0,0 +1,70 @@
+# Ami Minimum Age
+
+This document describes a design for a new feature to allow users to specify a minimum age for AMIs.
+
+(a @jonathan-innis's idea, see [this comment](https://github.com/aws/karpenter-provider-aws/issues/5382#issuecomment-1868068193))
+
+## Background
+
+Karpenter allows users to specify TTL for nodes, which configures the maximum age of nodes in the cluster. 
+
+This is useful to ensure that nodes are replaced periodically, and helps to ensure that the cluster is running on the latest AMIs available, 
+simplifying and automating the security patching process.
```

Comment:
One thing i'm curious about is that you're looking to use this in response to the AMI issue that you've tagged below. What would you have set on this amiMinimumAge if it was specified?

One of the issues with this is if you have a new AMI released as a CVE patch, or a roll forward from a faulty image, would you still want to wait the amiMinimumAge to get that AMI? I think in almost all cases like that, you'd want to grab the image as soon as possible.

Reply:
I think we would pick an interval that will comply with the most stringent security SLA that our company has defined, something like 5d.

We sourced the referred bad AMI in our production clusters the same day it was published. If we had this mechanism we would have pulled it in only in our integration environment, where we would (hopefully) detected the issue, and stopped the production clusters from pulling the new AMI until the problem was solved.

In the scenarios you're describing (an urgent fix-forward AMI release) we would have released another version of the EC2NodeClass, maybe with no or a shorter minimum age. That would have triggered a drift and reclaimed the nodes out of compliance (please correct me if I'm wrong).

this is correlated with [this comment](https://github.com/aws/karpenter-provider-aws/pull/5551#discussion_r1481936955) about how a new AMI and drift disruption work. I think I might not have a clear picture of it.